### DISCUSSION TITLE
Is Koka still under active development?

### DISCUSSION THREAD

**Discussion by woojamon** [koka-lang/koka#339] [General]
I love the ideas behind this language, but its been over a year without any commits on master. Is Koka still alive?

**Comment by daanx** [comment]
Hi @woojamon -- yes! Currently the active development is in `dev-fbip`, see also our recent tech report: 
<https://www.microsoft.com/en-us/research/publication/fp2-fully-in-place-functional-programming/>
We hope to put out a fresh release after the POPL deadline (Jul 11) that will include the new `fip` keyword and other goodies.

Now, having said this, as the website says, Koka is still experimental -- the main things lacking from a language perspective (in my opinion) is a system of dynamic overloading and integration with libuv; both are high on the list as we hope to ramp up Koka to be used for "real". I know it takes time but in contrast to many other language attempts, we really try to develop Koka from solid theoretical foundations and that takes some time; especially with the many new innovations in Koka like algebraic effect handlers and Perceus. We also spend a lot of time on engineering the runtime system to be of high quality.

**Reply by woojamon** [reply]
Thanks @daanx ! That's what I like most about Koka, is that its attempting to be built on solid theoretical foundations, not just some things that some people think would be nice/cool/interesting features. Also, being backed by Microsoft, has the potential to really go somewhere. Thanks for chiming in and I'll keep an eye on Koka!
